### PR TITLE
VH-82 Improvements & Fixes

### PR DESCRIPTION
**Pull request by rdgfuentes**
- [x] Deep linking searches page
- [x] Navigate back on vessel page profile
- [x] Parsing dates using UTC
- [x] Removed console.log
- [x] Refactored InfoFieldHistoric to display the modal in it
- [ ] Reuse VesselInfo interface to populate data coming from any source at loading time (before render) - To be addressed in a future PR
- [x] Fix console warnings on compilation
- [x] Fix failing tests

### CODE REVIEW

File: applications/vessel-history/src/classes/vessel.class.ts [hunk 1]
```diff
@@ -40,31 +47,41 @@ export class VesselInfo {
     }
     const dataValues =
       tmtValue?.map((historicValue: AnyValueList) => {
-        console.log(historicValue.firstSeen)
         const value: HistoricValue = {
           data: historicValue.value,
-          start: historicValue.firstSeen ? new Date(historicValue.firstSeen) : null,
-          end: historicValue.endDate ? new Date(historicValue.endDate) : null,
+          start: historicValue.firstSeen
+            ? DateTime.fromISO(historicValue.firstSeen, { zone: 'UTC' })
```
Comment (j8seangel) [inline applications/vessel-history/src/classes/vessel.class.ts:53]: Seems we have this function repeated in several places, what about creating a helper like `isoToUtcDate` to reduce repetition?

File: applications/vessel-history/src/features/home/Home.tsx [hunk 2]
```diff
@@ -48,13 +51,13 @@ const Home: React.FC<LoaderProps> = (): React.ReactElement => {
...
     setVessels([])
-    if (query.length >= minimumCharacters) {
+    if (query?.length >= minimumCharacters) {
       fetchData(query)
     }
   }, [query, fetchData])
 
   const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setQuery(e.target.value)
+    dispatchQueryParams({ q: e.target.value })
```
Comment (j8seangel) [inline applications/vessel-history/src/features/home/Home.tsx:60]: nice 👍 !

File: applications/vessel-history/src/routes/routes.selectors.ts [hunk 3]
```diff
@@ -45,7 +51,7 @@ export const selectVesselProfileId = createSelector(
   }
 )
 
-export const selectQueryParam = <T = any>(param: WorkspaceParam) =>
```
Comment (j8seangel) [inline applications/vessel-history/src/routes/routes.selectors.ts:48]: isn't this generic util anymore?

**Review by j8seangel — COMMENTED** [review]
> Fix console warnings on compilation

thanks! 🙇 

Tested locally and it is working great, I'd just add some hint of these elements are clickables:
image to open the modal and see the details.

Great job and the test looks good to ensure we don't push sources to production:
```
jest.mock('data/constants', () => ({
  get SHOW_VESSEL_API_SOURCE() {
    return mockShowSourceAPI() // set some default value
  },
}))
```

File: applications/vessel-history/src/classes/vessel.class.ts (hunk 1 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/classes/vessel.class.ts:53]: Most of the `vessel.class.ts` will be changed in another branch `vessel-history/VH-82-refactor-thunks`

File: applications/vessel-history/src/routes/routes.selectors.ts (hunk 3 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/routes/routes.selectors.ts:48]: not for now

**Pull request merged**